src/Components/DeathReport/DeathReport.tsx

```diff
@@ -96,57 +95,55 @@ export default function PrintDeathReport(props: { id: string }) {
-            cause_of_death:
-              patientRes.data.last_consultation?.discharge_notes || "",
-            hospital_died_in: patientRes.data.last_consultation.facility_name,
-            date_declared_positive: patientRes.data.date_declared_positive
-              ? dayjs(patientRes.data.date_declared_positive).toDate()
-              : "",
-            date_of_admission: patientRes.data.last_consultation.admission_date
-              ? dayjs(patientRes.data.last_consultation.admission_date).toDate()
-              : "",
-            date_of_test: patientRes.data.date_of_test
-              ? dayjs(patientRes.data.date_of_test).toDate()
-              : "",
-            date_of_result: patientRes.data.date_of_result
-              ? dayjs(patientRes.data.date_of_result).toDate()
-              : "",
-            date_of_death: patientRes.data.last_consultation.death_datetime
-              ? dayjs(patientRes.data.last_consultation.death_datetime).toDate()
-              : "",
-          };
-          setPatientData(data);
-        }
-        setIsLoading(false);
-      }
-    },
-    [dispatch, id]
-  );
+  const {
+    res,
+    data: Rpatient,
+    loading: _isLoading,
+    refetch,
+  } = useQuery(routes.getPatient, { pathParams: { id } });
+
+  useEffect(() => {
+    if (res?.ok && Rpatient) {
+      setPatientName(Rpatient.name);
+      const patientGender = getPatientGender(Rpatient);
+      const patientAddress = getPatientAddress(Rpatient);
+      const patientComorbidities = getPatientComorbidities(Rpatient);
+      const data = {
+        ...Rpatient,
+        gender: patientGender,
+        address: patientAddress,
+        comorbidities: patientComorbidities,
+        is_declared_positive: Rpatient.is_declared_positive ? "Yes" : "No",
+        is_vaccinated: patientData.is_vaccinated ? "Yes" : "No",
+        cause_of_death: Rpatient.last_consultation?.discharge_notes || "",
+        hospital_died_in: Rpatient.last_consultation?.facility_name,
+        date_declared_positive: Rpatient.date_declared_positive
+          ? dayjs(Rpatient.date_declared_positive).toDate()
+          : "",
+        date_of_admission: Rpatient.last_consultation?.admission_date
+          ? dayjs(Rpatient.last_consultation?.admission_date).toDate()
+          : "",
+        date_of_test: Rpatient.date_of_test
+          ? dayjs(Rpatient.date_of_test).toDate()
+          : "",
+        date_of_result: Rpatient.date_of_result
+          ? dayjs(Rpatient.date_of_result).toDate()
+          : "",
+        date_of_death: Rpatient.last_consultation?.death_datetime
+          ? dayjs(Rpatient.last_consultation?.death_datetime).toDate()
+          : "",
+      };
+      setPatientData(data);
+    }
+  }, [Rpatient]);
 
   useAbortableEffect(
     (status: statusType) => {
-      fetchpatient(status);
+      if (!status.aborted) {
+        refetch();
+      }
     },
-    [dispatch, fetchpatient]
+    [refetch]
```

Comment:
You can remove the useAbortableEffect when using useQuery. It'll handle aborts by itself.

Reply:
Removed entire useAbortableEffect